app/src/pages/playground/PlaygroundTemplate.tsx

```diff
@@ -117,3 +133,50 @@ function DeleteButton(props: PlaygroundInstanceProps) {
     </TooltipTrigger>
   );
 }
+
+type SaveButtonProps = {
+  instanceId: number;
+  setDialog: (dialog: React.ReactNode) => void;
+};
+
+function SaveButton({ instanceId, setDialog }: SaveButtonProps) {
+  const instance = usePlaygroundContext((state) =>
+    state.instances.find((instance) => instance.id === instanceId)
+  );
+  if (!instance) {
+    throw new Error(`Instance ${instanceId} not found`);
+  }
+  const prompt = instance.prompt;
+  const onSubmit: SavePromptSubmitHandler = useCallback(
+    (params) => {
+      // eslint-disable-next-line no-console
+      console.log("saving prompt", instanceId, params);
+    },
+    [instanceId]
+  );
+  const onSave = () => {
+    setDialog(
+      <Dialog title={prompt?.id ? "Update Prompt" : "Save Prompt"}>
+        <SavePromptForm onSubmit={onSubmit} />
+      </Dialog>
+    );
+  };
+  return (
+    <>
+      <TooltipTrigger delay={100}>
+        <TriggerWrap>
+          <Button
+            variant="default"
+            size="compact"
+            icon={<Icon svg={<Icons.SaveOutline />} />}
+            aria-label="Save this prompt"
+            onClick={onSave}
+          />
```

Comment:
I was debating if this should be variant primary to make it stand out.

Additionally, if it should maybe be variant "ghost" until the instance differs from the prompt and then switch to "primary" after, to imply that there are changes from the saved prompt